inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,145 @@ See the OMA book for examples.
 **Extra** Cool Exercise : Identify Patterns and Clusters
 To make the exercise more engaging, let's do an exploratory data analysis task:
 
-#### 1. Cluster Analysis on PCoA/PCA Results:
+### 1: k-means Clustering on Reduced Dimensions and Visualization
 
-i. Perform clustering on the ordination results (e.g., k-means clustering).
-ii. Visualize the clusters on the ordination plots.
+- Performs k-means clustering on reduced dimensional data obtained from PCoA.
+- Visualizes the clusters in a scatter plot of the reduced dimensions.
 
 ```{r}
-# Setting seed for reproducibility
-set.seed(123)
-
-# Perform k-means clustering on the reduced dimensions of pcoa_bray
-pcoa_bray_clusters <- kmeans(reducedDim(pcoa_bray), centers = 3)$cluster
-
-# Plot the reduced dimensions of pcoa_bray with clusters as colors
-plotReducedDim(pcoa_bray, colour_by = pcoa_bray_clusters)
-
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+#| label: heatmap-visualization-ex3
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
+library(ggplot2)
+library(cluster)
+
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+se <- SummarizedExperiment(assays = list(counts = assay(GlobalPatterns)),
+                           colData = colData(GlobalPatterns),
+                           rowData = rowData(GlobalPatterns))
+
+# Perform PCoA on the abundance data
+distance_matrix <- vegdist(t(assay(se)), method = "bray")
+pcoa_results <- cmdscale(distance_matrix, k = 2)
+pcoa_data <- data.frame(pcoa_results)
+colnames(pcoa_data) <- c("PCoA1", "PCoA2")
```

Comment:
Instead, use runMDS

Reply:
I am facing a continuous error which indicates that the runMDS function is not available in the mia package, could you help me with me with it?

Reply:
Try scater::runMDS()